internal/controller/controllers/agentclusterinstall_controller.go

```diff
@@ -18,10 +18,14 @@ package controllers
 
 import (
 	"context"
+	"encoding/json"
```

Comment:
This is not the right controller.
You should use internal/controller/controllers/clusterdeployments_controller.go